src/System.Drawing.Common/src/System/Drawing/Drawing2D/GraphicsPath.Windows.cs

```diff
@@ -183,46 +183,36 @@ public FillMode FillMode
             }
         }
 
-        private PathData _GetPathData()
+        private unsafe PathData _GetPathData()
         {
-            int ptSize = Marshal.SizeOf(typeof(GPPOINTF));
+            int count = PointCount;
 
-            int numPts = PointCount;
+            PathData pathData = new PathData()
+            {
+                Types = new byte[count],
+                Points = new PointF[count]
+            };
 
-            PathData pathData = new PathData() { Types = new byte[numPts] };
+            if (count == 0)
+                return pathData;
 
-            IntPtr memoryPathData = Marshal.AllocHGlobal(3 * IntPtr.Size);
-            IntPtr memoryPoints = Marshal.AllocHGlobal(checked(ptSize * numPts));
-            try
+            fixed (byte* t = pathData.Types)
+            fixed (PointF* p = pathData.Points)
             {
-                GCHandle typesHandle = GCHandle.Alloc(pathData.Types, GCHandleType.Pinned);
-                try
+                GpPathData data = new GpPathData
                 {
-                    IntPtr typesPtr = typesHandle.AddrOfPinnedObject();
+                    Count = count,
+                    Points = p,
+                    Types = t
+                };
```

Comment:
Doesn't this have to be fixed as well?

Reply:
> or is it guaranteed to be on stack and hence not moving

Correct.  The language won't even let you use fixed here; it'll produce an error something about not letting you use fixed with an already fixed expression.